src/app/loadout-builder/mod-assignments.ts

```diff
@@ -0,0 +1,140 @@
+import { DestinyEnergyType } from 'bungie-api-ts/destiny2';
+import _ from 'lodash';
+import { DimItem, PluggableInventoryItemDefinition } from '../inventory/item-types';
+import { ProcessItem, ProcessMod } from './process-worker/types';
+
+const DEFAULT_ENERGY_INVESTMENT = 1000;
+
+interface ItemAssignments {
+  mods: ProcessMod[];
+  energySwapped: boolean;
+}
+
+export class ModAssignments {
```

Comment:
Would you mind documenting the class?

In general I do try to avoid classes, especially mutable classes, largely because they can't safely pass between main process and worker, or be serialized/deserialized to IndexedDB.

Reply:
Yeah I normally do too, I kinda saw this as a similar problem to the set tracker we have in the process thing. Basically we want to keep a single copy of something and apply mutable changes on it. Ideally I want to split this whole thing so we aren't needing to convert items to process items just for figuring out the mods. It would mean we need two versions of that algorithm but I think that will be fine if I pull some of the logic out into helper methods that can handle both types.